src/styles/foundation/duration.scss

```diff
@@ -1,24 +1,19 @@
-$unit: 100ms;
+@import '../polaris-tokens/duration.map';
 
-$duration-data: (
-  fast: $unit,
-  base: $unit * 2,
-  slow: $unit * 3,
-  slower: $unit * 4,
-  slowest: $unit * 5,
-);
+$duration-data: $polaris-duration-map;
 
 /// Returns the duration value for a given variant.
 ///
 /// @param {String} $variant - The key for the given variant.
-/// @return {Number} The spacing for the variant (in miliseconds).
+/// @return {Number} The duration for the variant (in miliseconds).
 
 @function duration($variant: base) {
-  $fetched-value: map-get($duration-data, $variant);
+  $interpolated-value: 'duration-' + $variant;
```

Comment:
I guess the only question that remains is whether or not we want to keep these _"foundation sass functions"_ consistent and always consume a `token.token-map.scss` format vs the `token.map.scss` format I am currently using.

In the solution presented above, I have to "interpolate" the _"token prefix"_ in order to keep the same consumer API (`color: token('variant');`) – otherwise, we would need to pass in the full name of the key. Alternatively, you can forgo the custom function and just use `map-get`:
`transition-duration: map-get($polaris-duration-map, 'duration-tight');`

Using `spacing` as an example, here is the difference:

**spacing.map.scss**
```scss
$polaris-spacing-map: (
  'spacing-none': (
    0,
  ),
  'spacing-extra-tight': (
    4px,
  ),
  // ...
);
```

**spacing.spacing-map.scss**
```scss
$polaris-spacing: (
  'none': 0,
  'extra-tight': 4px,
  // ...
);
```

Does anyone have an opinion on this? If we want the `token.token-map.scss` format, we will need to do another `polaris-tokens` beta.

Reply:
Not sure I follow entirely. Do you know why we provide both?

Reply:
@kaelig and I made an executive decision to stop providing the snowflake `{token}.{token}-map.scss` style map.

Read my in-depth comment here:
https://github.com/Shopify/polaris-tokens/pull/54#issuecomment-481746556

In our opinion, we should remove this extra build step from `polaris-tokens` to produce the "snowflake" map and go all in on providing just the `{token}.map.scss`. Future token maps will require Polaris to adjust their "property function" slightly, but thats a trivial task and upgraders can just look at this PR for guidance!